src/DataProtection/Azure.Storage.Blob/src/AzureDataProtectionBuilderExtensions.cs

```diff
@@ -0,0 +1,102 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Azure.Storage;
+using Azure.Storage.Blobs;
+using Microsoft.AspNetCore.DataProtection.Azure.Storage.Blob;
+using Microsoft.AspNetCore.DataProtection.KeyManagement;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.DataProtection
+{
+    /// <summary>
+    /// Contains Azure-specific extension methods for modifying a
+    /// <see cref="IDataProtectionBuilder"/>.
+    /// </summary>
+    public static class AzureDataProtectionBuilderExtensions
+    {
+        /// <summary>
+        /// Configures the data protection system to persist keys to the specified path
+        /// in Azure Blob Storage.
+        /// </summary>
+        /// <param name="builder">The builder instance to modify.</param>
+        /// <param name="blobUri">The full URI where the key file should be stored.
+        /// The URI must contain the SAS token as a query string parameter.</param>
+        /// <returns>The value <paramref name="builder"/>.</returns>
+        /// <remarks>
+        /// The container referenced by <paramref name="blobUri"/> must already exist.
+        /// </remarks>
+        public static IDataProtectionBuilder PersistKeysToAzureBlobStorage(this IDataProtectionBuilder builder, Uri blobUri)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+            if (blobUri == null)
+            {
+                throw new ArgumentNullException(nameof(blobUri));
+            }
+
+            var uriBuilder = new BlobUriBuilder(blobUri);
+
+            // The SAS token is present in the query string.
+
+            if (string.IsNullOrEmpty(uriBuilder.Query))
+            {
+                throw new ArgumentException(
+                    message: "URI does not have a SAS token in the query string.",
+                    paramName: nameof(blobUri));
+            }
+
+            var credentials = new StorageSharedKeyCredential(uriBuilder.AccountName, uriBuilder.Query);
+            var blobName = uriBuilder.BlobName;
+            uriBuilder.Query = null; // no longer needed
+            var blobAbsoluteUri = uriBuilder.ToUri();
+
+            var client = new BlobContainerClient(blobAbsoluteUri, credentials);
```

Comment:
Just pass the `blobUri` to `BlobContainerClient` ctor. It should work 🤞

Reply:
I think I'm wrong. What this overload should do is to see if Uri is SAS uri and if not use DefaultAzureCredential for authentication.

Reply:
To check if it's a SAS url create a `bloburibuilder` and check the https://docs.microsoft.com/en-us/dotnet/api/azure.storage.blobs.bloburibuilder.sas?view=azure-dotnet#Azure_Storage_Blobs_BlobUriBuilder_Sas property